test/functional/collection_tests.js

```diff
@@ -1898,4 +1898,27 @@ describe('Collection', function() {
       }
     });
   });
+
+  it('should not allow atomic operators on findOneAndReplace', function(done) {
```

Comment:
this is a good example of a test we should unit test. This is creating a client, connecting to a server(s) and sending a command, all to validate that we check inputs before the command is sent. Not a bad issue in isolation, but can really add up over hundred/thousands of tests.

Also, this file has a lot of duplication that I imagine we can reduce. Every test wants to set up a client, get a collection, and then call some method on the collection. We can probably DRY those first parts. Patrick did a great job of this over on the swift driver's [sessions tests](https://github.com/mongodb/mongo-swift-driver/blob/master/Tests/MongoSwiftTests/ClientSessionTests.swift#L35) for example.

Reply:
I made NODE-1986 to fix this file.